tools/run_cargo_fmt.sh

```diff
@@ -6,6 +6,28 @@
 # other use of rustfmt on the machine
 RUSTFMT_VERSION=0.7.1
 
+# Format overwrites changes, which is probably good, but it's nice to see
+# what it has done
+#
+# `git status --porcelain` formats things for scripting
+# | M changed file, unstaged
+# |M  changed file, staged (git add has run)
+# |MM changed file, some staged and some unstaged changes (git add then changes)
+# |?? untracked file
+if git status --porcelain | grep '^.M'; then
```

Comment:
What about

    if git status --porcelain | grep '^.M.*\.rs' > /dev/null; then

That checks that the changed files are .rs files only, and hides the git status before the actual message.